konrad/ozone.py

```diff
@@ -147,6 +148,86 @@ def __call__(self, atmosphere, convection, **kwargs):
         )
 
 
+class OzoneRedistributed(Ozone):
+    """Ozone redistribution following method by Hardiman et al. (2019)"""
+    def __init__(self):
+        self._f = None
+    
+    def get_tropopause_index(self, atmosphere):
+        """Get thermal and ozone tropopause indexes"""
+        z = atmosphere["z"][0, :]/1000
+        T = atmosphere["T"][0, :]
+        lapse_rate_K_per_km = np.gradient(T, z)
+        
+        mask1 = np.where(lapse_rate_K_per_km > -2)[0]
+        thermal_tropopause = False
+        i=0
+        while not thermal_tropopause:
+            index = mask1[i]
+            limit_test = np.where(z-z[index] >= 2)[0][0]
+            region = lapse_rate_K_per_km[index:limit_test + 1]
+            if np.all(region >= -2):
+                thermal_tropopause = index
+            i += 1
```

Comment:
Just a general note: You could consider moving this into a separate method of the atmosphere component, e.g. `Atmosphere.get_thermal_tropopause()`. But this is more a hint on how one _could_ structure functionality ;)

Reply:
I did not want to alter the `Atmosphere` class. When I have corrected this PR, we review it again and it comes out I will come to that point.